src/sage/rings/function_field/drinfeld_modules/morphism.py

```diff
@@ -404,3 +419,365 @@ def ore_polynomial(self):
             True
         """
         return self._ore_polynomial
+
+    @coerce_binop
+    def __add__(self, other):
+        r"""
+        Return the sum of this morphism and ``other``.
+
+        INPUT:
+
+        - ``other`` -- a morphism of Drinfeld modules in the
+          same category
+
+        TESTS::
+
+            sage: Fq = GF(5)
+            sage: A.<T> = Fq[]
+            sage: K.<z> = Fq.extension(3)
+            sage: phi = DrinfeldModule(A, [z, 0, 1, z])
+            sage: t = phi.ore_variable()
+            sage: f = phi.hom(t + 1)
+            sage: g = T * f
+            sage: f + g  # indirect doctest
+            Drinfeld Module morphism:
+              From: Drinfeld module defined by T |--> z*t^3 + t^2 + z
+              To:   Drinfeld module defined by T |--> (2*z^2 + 4*z + 4)*t^3 + (3*z^2 + 2*z + 2)*t^2 + (2*z^2 + 3*z + 4)*t + z
+              Defn: (2*z^2 + 4*z + 4)*t^4 + (z + 1)*t^3 + t^2 + (2*z^2 + 4*z)*t + z + 1
+
+        We check that coercion from the function ring works::
+
+            sage: F = phi.frobenius_endomorphism()
+            sage: F + T
+            Endomorphism of Drinfeld module defined by T |--> z*t^3 + t^2 + z
+              Defn: (z + 1)*t^3 + t^2 + z
+
+        """
+        u = self.ore_polynomial() + other.ore_polynomial()
```

Comment:
Any particular reason an intermediate variable is used here but not elsewhere? Might be a good idea to approach this consistently.